presto-password-authenticators/src/main/java/io/prestosql/plugin/password/filebasedauthenticator/FileBasedAuthenticator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.filebasedauthenticator;
+
+import com.google.common.collect.ImmutableSet;
+import io.airlift.log.Logger;
+import io.prestosql.spi.security.AccessDeniedException;
+import io.prestosql.spi.security.BasicPrincipal;
+import io.prestosql.spi.security.PasswordAuthenticator;
+
+import javax.inject.Inject;
+import javax.naming.AuthenticationException;
+
+import java.io.IOException;
+import java.security.Principal;
+
+import static io.prestosql.plugin.password.filebasedauthenticator.Users.readUsers;
+
+public class FileBasedAuthenticator
+        implements PasswordAuthenticator
+{
+    private static final Logger log = Logger.get(FileBasedAuthenticator.class);
+
+    private Users users;
+
+    @Inject
+    public FileBasedAuthenticator(FileBasedAuthenticationConfig config)
+    {
+        Users users;
+        try {
+            users = readUsers(config.getConfigFile());
+        }
+        catch (IOException e) {
+            log.warn(String.format("Could not read config file %s, no queries will go through", config.getConfigFile()));
```

Comment:
No need for `format()` since Airlift logger takes format strings directly. Also, we should log the original exception, so the administrator has a chance of telling what went wrong:
```java
log.error(e, "Failed to read password database: %s", config.getConfigFile());
```

Reply:
Also, it might be better to simply throw here, rather than leaving the plugin and server in a broken state.